Update 025 deg RYF configuration with New Grid, Topography, and Initial Conditions

ezhilsabareesh8 opened the pull request:
**1. Summary:**
This PR updates the 0.25 deg RYF configuration with the following improvements:

- A new grid file with zonal C-grid points on the equator and an extended Mercator region down to 75°S.
- Updated topography generated from the GEBCO 2024 dataset, ensuring compatibility with C-grid connectivity requirements.
- Initial conditions derived from the WOA2023 dataset.
- Surface-level salinity restoring based on the re-gridded WOA2023 dataset.
- A vertical grid with 75 levels.

**2. Issues Addressed:**
https://github.com/COSIMA/access-om3/issues/172
https://github.com/COSIMA/access-om3/issues/161
https://github.com/COSIMA/access-om3/issues/249
https://github.com/COSIMA/access-om3/issues/207

**8. Formatting**
Changes to MOM_input are in the same order as docs/MOM_input

anton-seaice reviewed (commented):
Thanks @ezhilsabareesh8 

/g/data/vk83/prerelease/configurations/inputs/access-om3/mom/initial_conditions/global.025deg/2025.01.30/ocean_temp_salt.res.nc

does not have much metadata - was there a `finalise.sh` step expected that was forgotten ?

Also - as noted on zulip - there's only 38 levels in the file (but should be 75). We should make an issue to fix it. (and finalise https://github.com/ACCESS-NRI/initial_conditions_access-om3/pull/1 before a formal release)

anton-seaice commented:
!test repro commit

jo-basevi commented:
> model-config-tests is trying to download spack.location from the wrong location

Yes it is using the wrong url.. I think the url changed with the new version `v4` of `build-cd` which supports multi-HPC deployments. So I'll work on getting that test updated

CodeGat commented:
Sorry about that. The file is now `SUPERCOMPUTER.spack.location` (eg. `Gadi.spack.location`)

ezhilsabareesh8 commented:
> does not have much metadata - was there a `finalise.sh` step expected that was forgotten ?
> 
> Also - as noted on zulip - there's only [38 levels in the file (but should be 75)](https://github.com/COSIMA/access-om3/issues/300). (and finalise [ACCESS-NRI/initial_conditions_access-om3#1](https://github.com/ACCESS-NRI/initial_conditions_access-om3/pull/1) before a formal release)

I have fixed the [issue](https://github.com/COSIMA/access-om3/issues/300) of number of levels in the initial condition file  and added the metadata in the initial condition file. The new file is in `/g/data/vk83/prerelease/configurations/inputs/access-om3/mom/initial_conditions/global.025deg/2025.03.19`, which has 76 levels same as vgrid. I will update this PR once the test runs are okay.

ezhilsabareesh8 commented:
> I will update this PR once the test runs are okay.

Test runs are fine.

anton-seaice commented:
@ezhilsabareesh8 - you accidentally commited the restart.yaml - can you drop those please ?

and then do the test repro commit command :)

ezhilsabareesh8 commented:
!test repro commit

ezhilsabareesh8 commented:
!test repro https://github.com/ACCESS-NRI/access-om3-configs/commit/ce75d880af94d249bf6a0b0e28daf8f43a0d2941

anton-seaice commented:
`!test repro commit` was right - it think just dont put a newline at the end

ezhilsabareesh8 commented:
!test repro commit

anton-seaice commented:
QA tests are failing due to https://github.com/ACCESS-NRI/model-config-tests/issues/132 - we can probably force this to (squash) merge by someone with the right permissions.

dougiesquire commented:
The failing QA test is not preventing merging. It just needs an approving review. Or am I misunderstanding what you're saying?

anton-seaice commented:
!test repro commit

(the last one timed out!)

anton-seaice commented:
!test repro commit

dougiesquire commented on config.yaml:
```diff
@@ -20,18 +20,19 @@ model: access-om3
 input:
-    - /g/data/vk83/configurations/inputs/access-om3/share/meshes/global.025deg/2024.03.26/access-om2-025deg-ESMFmesh.nc
-    - /g/data/vk83/configurations/inputs/access-om3/share/meshes/global.025deg/2024.03.26/access-om2-025deg-nomask-ESMFmesh.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/cice/grids/global.025deg/2025.01.30/grid.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/cice/grids/global.025deg/2025.02.17/kmt.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/mom/grids/mosaic/global.025deg/2025.01.30/ocean_hgrid.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/mom/grids/vertical/global.025deg/2025.03.12/ocean_vgrid.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/mom/initial_conditions/global.025deg/2025.03.19/ocean_temp_salt.res.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/mom/surface_salt_restoring/global.025deg/2025.01.30/salt_sfc_restore.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/share/grids/2025.02.17/topog.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/share/meshes/2025.02.17/access-om3-025deg-ESMFmesh.nc
+    - /g/data/vk83/prerelease/configurations/inputs/access-om3/share/meshes/2025.02.17/access-om3-025deg-nomask-ESMFmesh.nc
```
A really minor point and definitely not needed for this PR, but some of these input files are derived from other input files (the cice grids and meshes). In these cases, we should try to generate the input from the files in `/g/data/vk83` to make the provenance a bit easier to parse.

It's not immediately clear for example that the meshes are created from same the `ocean_hgrid` files used in this configuration:

```
$ ncdump -hsc /g/data/vk83/prerelease/configurations/inputs/access-om3/share/meshes/2025.02.17/access-om3-025deg-ESMFmesh.nc
netcdf access-om3-025deg-ESMFmesh {
.
.
.
// global attributes:
		:gridType = "unstructured mesh" ;
		:timeGenerated = "2025-02-17 14:13:05.912107" ;
		:created_by = "as2285" ;
		:inputFile = "/g/data/tm70/as2285/make_OM3_025deg_topo/ocean_hgrid.nc (md5 hash: c65fea49f3a94530a36c53164b316dd2), /g/data/tm70/as2285/make_OM3_025deg_topo/ocean_mask.nc (md5 hash: 427b8fe00506dfbda20b74f8fad8fd73)" ;
		:history = "Created by as2285 (anton-seaice) on 2025-02-17, using https://github.com/COSIMA/om3-scripts/blob/92cc161e1f19dfa04182d03096591de35f8bbd3d/mesh_generation/generate_mesh.py: python3 generate_mesh.py --grid-type=mom --grid-filename=/g/data/tm70/as2285/make_OM3_025deg_topo/ocean_hgrid.nc --mesh-filename=/g/data/tm70/as2285/make_OM3_025deg_topo/access-om3-025deg-ESMFmesh.nc --mask-filename=/g/data/tm70/as2285/make_OM3_025deg_topo/ocean_mask.nc --wrap-lons" ;
		:_NCProperties = "version=2,netcdf=4.9.2,hdf5=1.14.3" ;
		:_SuperblockVersion = 2 ;
		:_IsNetcdf4 = 1 ;
		:_Format = "netCDF-4" ;
}
```

Sure, we can back it out from the md5sum, but for inputs for released configs it would be nice to have it clear in the paths.

dougiesquire commented on manifests/restart.yaml:
```diff
@@ -1,3 +1,3 @@
 format: yamanifest
 version: 1.0
---- {}
+--- {}
```
Please remove new line added at end of file.

ezhilsabareesh8 replied:
The editor is not showing any new line at the end of file

dougiesquire reviewed (commented):
Thanks @ezhilsabareesh8, looks great! I've one very minor change and some pedantry.

anton-seaice reviewed (approved):
Thanks @ezhilsabareesh8 - looks good

anton-seaice commented:
@CodeGat - why are QA CI Checks skipped ? They were failing until the bot committed ...

anton-seaice commented:
They were skipped because of this line:

https://github.com/ACCESS-NRI/model-config-tests/blob/22e2cd18bf4475dff1f90a499f1f7afba7a7c93a/.github/workflows/config-pr-1-ci.yml#L158

However this is problematic, because the "skipped" result overwrote the previous "failed" for this check ... :-)

dougiesquire reviewed (changes requested):
There're a few changes to the `MOM_input` required for consistency with `MOM_parameter_doc.short`:

```diff
@@ -181,7 +157,7 @@ ALE_COORDINATE_CONFIG = "FILE:ocean_vgrid.nc,interfaces=zeta" ! default = "UNIFO
                                 !                the filename and two variable names, separated
                                 !                by a comma or space, for sigma-2 and dz. e.g.
                                 !                HYBRID:vgrid.nc,sigma2,dz
-!ALE_RESOLUTION = 2.303499698638916, 2.6903486251831055, 3.1421399116516113, 3.6697616577148438, 4.285917282104492, 5.005424499511719, 5.845563888549805, 6.826459884643555, 7.971549987792969, 9.308074951171875, 10.867660522460938, 12.686931610107422, 14.808158874511719, 17.279945373535156, 20.157821655273438, 23.504684448242188, 27.390975952148438, 31.894271850585938, 37.097900390625, 43.088226318359375, 49.94970703125, 57.757049560546875, 66.56375122070312, 76.386962890625, 87.18865966796875, 98.85760498046875, 111.1953125, 123.914794921875, 136.6578369140625, 149.03271484375, 160.6646728515625, 171.2481689453125, 180.5816650390625, 188.5797119140625, 195.2608642578125, 200.720703125, 205.10205078125, 208.565185546875, 211.2705078125, 213.363525390625, 214.97119140625, 216.198974609375, 217.13232421875, 217.83984375, 218.37451171875, 218.7783203125, 219.08203125, 219.310546875, 219.482421875, 219.6123046875 !   [m]
+!ALE_RESOLUTION = 1.0825614929199219, 1.1963462829589844, 1.322089672088623, 1.4610481262207031, 1.614609718322754, 1.784308910369873, 1.9718408584594727, 2.1790781021118164, 2.4080896377563477, 2.661160469055176, 2.940814971923828, 3.249845504760742, 3.591329574584961, 3.968667984008789, 4.385614395141602, 4.846321105957031, 5.355350494384766, 5.917766571044922, 6.539115905761719, 7.225547790527344, 7.983818054199219, 8.821372985839844, 9.746376037597656, 10.767845153808594, 11.895652770996094, 13.140586853027344, 14.514511108398438, 16.030319213867188, 17.7020263671875, 19.544876098632812, 21.575271606445312, 23.810821533203125, 26.270294189453125, 28.973419189453125, 31.94091796875, 35.194000244140625, 38.75390625, 42.641632080078125, 46.876739501953125, 51.476593017578125, 56.45489501953125, 61.82025146484375, 67.5743408203125, 73.70965576171875, 80.207763671875, 87.03759765625, 94.1534423828125, 101.4951171875, 108.9879150390625, 116.5452880859375, 124.0714111328125, 131.4671630859375, 138.6346435546875, 145.484130859375, 151.938232421875, 157.93701171875, 163.439697265625, 168.42431640625, 172.8876953125, 176.842041015625, 180.3125, 183.33154296875, 185.938720703125, 188.175048828125, 190.08251953125, 191.701171875
 REMAPPING_SCHEME = "PPM_H4"     ! default = "PLM"
                                 ! This sets the reconstruction scheme used for vertical remapping for all
                                 ! variables. It can be one of the following schemes:
@@ -372,7 +348,7 @@ VISC_REM_BUG = False            !   [Boolean] default = True
                                 ! VISC_REM_BT_WEIGHT_BUG in MOM_barotropic.

 ! === module MOM_continuity_PPM ===
-ETA_TOLERANCE = 1.0E-06         !   [m] default = 2.5E-09
+ETA_TOLERANCE = 1.0E-06         !   [m] default = 3.75E-09
                                 ! The tolerance for the differences between the barotropic and baroclinic
                                 ! estimates of the sea surface height due to the fluxes through each face.  The
                                 ! total tolerance for SSH is 4 times this value.  The default is
@@ -611,10 +587,9 @@ HBD_REMAPPING_USE_OM4_SUBCELLS = False !   [Boolean] default = True
                                 ! this option to false.

 ! === module MOM_sum_output ===
-MAXTRUNC = 10000                ! [truncations save_interval-1] default = 0
-                                ! The run will be stopped, and the day set to a very
-                                ! large value if the velocity is truncated more than
-                                ! MAXTRUNC times between energy saves.  Set MAXTRUNC to 0
+MAXTRUNC = 10000                !   [truncations save_interval-1] default = 0
+                                ! The run will be stopped, and the day set to a very large value if the velocity
+                                ! is truncated more than MAXTRUNC times between energy saves.  Set MAXTRUNC to 0
                                 ! to stop if there is any truncation of velocities.

 ! === module ocean_stochastics_init ===
```

CodeGat commented:
Yeah currently we don't do the QA checks from bot accounts either, but we can change this quite easily. I can do that now, if you'd like!

dougiesquire reviewed (approved):
Thanks @ezhilsabareesh8. I fixed up the added newline and another typo you introduced, I hope you don't mind.

LGTM - thanks for this!

anton-seaice commented:
@dougiesquire - yes - please turn on require checks to pass before merging :)
